tests/test_algorithm.py

```diff
@@ -1284,13 +1038,13 @@ def make_equity_minute_bar_data(cls):
             yield sid, create_minute_df_for_asset(
                 cls.trading_calendar,
                 cls.data_start,
-                cls.sim_params.end_session,
+                cls.END_DATE,
```

Comment:
Many of the `make_*_bar_data` methods used `sim_params` to construct data, but since there's no ordering dependency between `WithSimParams` and `With*BarReader`, `cls.sim_params` isn't guaranteed to exist when this method is called.  The right thing to do here is usually to use the calendar, which is a direct dependency of `With*BarReader`, and is therefore guaranteed to be initialized by the time this method is called.

Reply:
Thanks for cleaning this one up.